test/integration/testPutGet.js

```diff
@@ -657,3 +658,329 @@ describe('PUT GET test with GCS_USE_DOWNSCOPED_CREDENTIAL', function ()
+          });
+        },
+        function (callback)
+        {
+          // Run GET command
+          var statement = connection.execute({
+            sqlText: getQuery,
+            complete: function (err, stmt, rows)
+            {
+              var stream = statement.streamRows();
+              stream.on('error', function (err)
+              {
+                done(err);
+              });
+              stream.on('data', function (row)
+              {
+                assert.strictEqual(row.status, DOWNLOADED);
+                assert.strictEqual(row.size, fileSize);
+
+                // Decompress the downloaded file
+                var compressedFile = path.join(tmpDir,row.file);
+                var decompressedFile = path.join(tmpDir,'de-'+row.file);
+                const fileContents = fs.createReadStream(compressedFile);
+                const writeStream = fs.createWriteStream(decompressedFile);
+                const unzip = zlib.createGunzip();
+
+                fileContents.pipe(unzip).pipe(writeStream).on('finish', function() {
+                  // Verify the data of the downloaded file
+                  var data = fs.readFileSync(decompressedFile).toString();
+                  assert.strictEqual(data, ROW_DATA);
+                  fs.unlinkSync(compressedFile);
+                  fs.unlinkSync(decompressedFile);
+                })
+              });
+              stream.on('end', function (row)
+              {
+                callback();
+              });
+            }
+          });
+        },
+        function (callback)
+        {
+          // Remove files from staging
+          testUtil.executeCmd(connection, removeFile, callback);
+        },
+        function (callback)
+        {
+          // Drop temp table
+          testUtil.executeCmd(connection, dropTable, callback);
+        },
+        function (callback)
+        {       
+          fs.closeSync(tmpFile1.fd);
+          fs.unlinkSync(tmpFile1.name);
+          fs.closeSync(tmpFile2.fd);
+          fs.unlinkSync(tmpFile2.name);
+
+          // Delete the temporary folder
+          fs.rmdir(tmpDir, (err) =>
+          {
+            if (err) throw (err);
+          });
+          callback();
+        }
+      ],
+      done
+    );
+  });
+
+  it('testUploadMultifiles', function (done)
+  {
+    const count = 5;
+    var tmpFiles = [];
+    var createTable = `create or replace table ${TEMP_TABLE_NAME} (${COL1} STRING, ${COL2} STRING, ${COL3} STRING)`;
+    var removeFile = `REMOVE @${DATABASE_NAME}.${SCHEMA_NAME}.%${TEMP_TABLE_NAME}`;
+    var dropTable = `DROP TABLE IF EXISTS ${TEMP_TABLE_NAME}`;
+
+    // Create a tmp folder for downloaded files
+    var tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'get'));
+
+    var fileSize;
+
+    var getQuery = `GET @${DATABASE_NAME}.${SCHEMA_NAME}.%${TEMP_TABLE_NAME} file://${tmpDir}`;
+    // Windows user contains a '~' in the path which causes an error
+    if (process.platform == "win32")
```

Comment:
maybe let's create one method to generate proper file path depending on the platform?